### DISCUSSION TITLE
Is there any api available to get the colors from external toml file?

### DISCUSSION THREAD

**Discussion by dhruvinsh** [wezterm/wezterm#4217] [Q&A] [answered]
Hi for built-in color scheme we have `get_builtin_color_schemes` api. But for external toml files, do we have any api to load the colors? 

is the `wezterm.color.load_scheme` correct api for external toml colro scheme? I tried but it always fail to load the colors from file.

Error appears as `caused by: No such file or directory (os error 2)` but I am providing the absolute file path.

**Comment by wez** [comment]
https://wezfurlong.org/wezterm/config/lua/wezterm.color/load_scheme.html is what you should use.
re: "No such file or directory", check the syntax you used for the filename.
Note that lua is more strict about quoting backslashes in strings, so if you are on Windows, make sure you are using the correct quoting, or use forward slashes instead.
Note also that the flatpak version of wezterm runs in a container that may not have full filesystem access.

**Reply by dhruvinsh** [reply]
Thank you for the suggestion, I am a Mac user and I have installed the westerm with brew. 

In the permission, I have allowed the Full disk access for westerm. And I am using full path for toml color scheme, 
`local foo = wezterm.color.load_scheme("/home/ds/.config/wezterm/colors/nordfox.toml")` but still I get same error,

```console
callback error
stack traceback:
        [C]: in field 'load_scheme'
        /Users/ds/.config/wezterm/ui.lua:12: in main chunk
        [C]: in function 'require'
        [string "/Users/ds/.config/wezterm/wezterm.lua"]:2: in main chunk
caused by: No such file or directory (os error 2)
```

**Reply by nekowinston** [reply, marked as answer]
`/home/` seems wrong on macOS.

**Reply by wez** [reply]
Note that you can use https://wezfurlong.org/wezterm/config/lua/wezterm/home_dir.html to refer to your home directory

**Reply by dhruvinsh** [reply]
Oh yes, I don't know how I messed that up? yes you guys are right. I am away and I will try it later evening. I have windows system as well, I will give that a try though with correct escape sequence.

**Reply by dhruvinsh** [reply]
In windows system it works as expected, 
```lua
local ok
M.colorscheme = "nordfox"
ok, M.colors =
  pcall(wezterm.color.load_scheme, wezterm.home_dir .. "/.config/wezterm/colors/nordfox.toml")
if not ok then
  M.colors = wezterm.get_builtin_color_schemes()[M.colorscheme]
end
```

Update:
tried it in the MacOS as well. it works as well. `/home` was the error indeed.
